tests/VirtoCommerce.MarketingModule.Test/BestRewardPromotionPolicyTest.cs

```diff
@@ -65,6 +65,26 @@ public void EvaluateRewards_PickBestRelativeDiscount(int quantity, decimal expec
             Assert.Equal(expectedReward, rewards.First().Amount);
         }
 
+        [Fact]
+        public void EvaluatePromotion_GetBestPaymentReward()
+        {
+            //Agganre
+            var evalPolicy = GetPromotionEvaluationPolicy(GetPromotions("Get 5$ Off payment method PayPal"));
+            var productA = new ProductPromoEntry { ProductId = "ProductA", Price = 100, Quantity = 1 };
+            var context = new PromotionEvaluationContext
+            {
+                ShipmentMethodCode = "PayPal",
+                ShipmentMethodPrice = 5m,
```

Comment:
Looking at this (you are using ShipmentMethodCode instead of PaymentMethodCode), looks like the test checks something wrong, or even reward applies wrongly - reward should not be applied here, as there is not Payment method specified in the Context.
Also I would add expected results to the test method name, e.g. _Applied or smth

Reply:
need to change logic of the UnitTest